dffml/util/testing/manifest/shim.py

```diff
@@ -0,0 +1,1459 @@
+        setup_module_name = setup_path.stem
+        # Create module from file
+        spec = importlib.util.spec_from_file_location(
+            setup_module_name, setup_path
+        )
+        setup_module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(setup_module)
+        if not hasattr(setup_module, args.setup_function_name):
+            # Raise exception if the there is not setup function
+            raise ValueError(
+                f"setup module {args.setup!r} has no"
+                f" function named {args.setup_function_name!r}"
+            )
+        # Call the setup function
+        getattr(setup_module, args.setup_function_name)(**locals())
+    # Determine how to get the manifest
+    if args.input_action not in input_actions:
+        raise InputActionNotFound(
+            "Input action is used to read in manifest"
+            f" {args.input_action!r} not found in loaded input actions:"
+            f" {input_actions!r}"
+        )
+    input_action = input_actions[args.input_action]
+    # Get the manifest and any validation data that might be associated with it
+    contents = input_action(args)
+    # Validate the manifest
+    if not args.insecure:
+        if args.validation_action is None:
+            # Ensure we have a validation action if we are not in insecure mode
+            raise ValueError(
+                "If not in insecure mode a validation action must be specified."
+                f" Choose from {set(validation_actions.keys())!r}"
+            )
+        # Determine how to get validate the manifest
+        if args.validation_action not in validation_actions:
+            raise ValidationActionNotFound(
+                "Validation action is used to read in manifest"
+                f" {args.validation_action!r} not found in loaded"
+                f" validation actions: {validation_actions!r}"
+            )
+        validation_action = validation_actions[args.validation_action]
+        # Validate the manifest. Override unvalidated contents with just
+        # validated.
+        contents = validation_action(args, contents)
+    if args.only_validate:
+        # Bail if we are only validating the manifest and not parsing it
+        return
+    # Parse the manifest
+    manifest = parse(contents, parsers=parsers)
+    # Ensure we have
+    if "$schema" not in manifest:
+        raise SchemaNotFound(
+            "Manifest does not contain $schema key at document root"
+        )
+    # Determine document format and version
+    (
+        document_format,
+        document_version,
+        schema_url,
+        schema_validation_action,
+        schema_validation_target,
+    ) = parse_schema_url(manifest["$schema"])
+    # Validate schema
+    if not args.no_schema_check:
+        raise NotImplementedError(
+            f"schema validation not implemented {(schema_url, schema_validation_action, schema_validation_target)!r}"
+        )
```

Comment:
schema validation goes here

probably want to add `only_schema_check` here

Reply:
**TODO** validate `@context` for https://github.com/w3c/vc-data-model/issues/953

Reply:
https://multiformats.io/